data/crac/crac-util/src/main/java/com/farao_community/farao/data/crac_util/CracAliasesCreator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_util;
+
+import com.farao_community.farao.data.crac_api.*;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.*;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
+ */
+
+public class CracAliasesCreator {
+
+    private static final int POSITION_OF_SECOND_UCTE_NODE = 9;
+    private static final int BEGINNING_OF_ELEMENT_NAME = 17;
+
+    public CracAliasesCreator() {
+    }
+
+    public void createAliases(Crac crac, Network network) {
+        createAliases(crac, network, UcteNodeMatchingRule.ALL_8_CHARACTER_EQUAL);
+    }
+
+    public void createAliases(Crac crac, Network network, UcteNodeMatchingRule rule) {
+        // List (without duplicates) all the crac elements that need to be found in the network
+        Set<String> elementIds = new HashSet<>();
+        crac.getCnecs().forEach(cnec -> elementIds.add(cnec.getNetworkElement().getId()));
+        crac.getContingencies().forEach(contingency -> handleAliases(contingency.getNetworkElements(), elementIds));
+        crac.getNetworkActions().forEach(networkAction -> handleAliases(networkAction.getNetworkElements(), elementIds));
+        crac.getRangeActions().forEach(rangeAction -> handleAliases(rangeAction.getNetworkElements(), elementIds));
+
+        // Try to find a corresponding element in the network, and add elementId as an alias
+        elementIds.forEach(elementId -> {
+            Optional<Identifiable<?>> correspondingElement = network.getIdentifiables().stream().filter(identifiable -> anyMatch(identifiable, elementId, rule)).findAny();
+            correspondingElement.ifPresent(identifiable -> identifiable.addAlias(elementId));
+        });
+    }
+
+    private void handleAliases(Set<NetworkElement> networkElements, Set<String> elementIds) {
+        networkElements.forEach(networkElement -> elementIds.add(networkElement.getId()));
+    }
+
+    /* It only works correctly for Cnec with direction = both. This corrupts the other Cnec.
+    TODO : only look for the elements with reverse ID if all the thresholds of the cnec have a BOTH direction.
+     */
```

Comment:
Actually, this has to be fixed now!

Reply:
Will be done in another PR after https://github.com/farao-community/farao-core/pull/388